bitcoin/src/blockdata/transaction.rs

```diff
@@ -1084,23 +1092,27 @@ impl InputWeightPrediction {
     /// This is a `const` version of [`new`](Self::new) which only allows slices due to current Rust
     /// limitations around `const fn`. Because of these limitations it may be less efficient than
     /// `new` and thus is intended to be only used in `const` context.
-    pub const fn from_slice(input_script_len: usize, witness_element_lengths: &[usize]) -> Self {
+    pub const fn from_slice(input_script_len: u32, witness_element_lengths: &[u32]) -> Self {
         let mut i = 0;
-        let mut total_size = 0;
+        let mut total_size: u32 = 0;
         // for loops not supported in const fn
         while i < witness_element_lengths.len() {
             let elem_len = witness_element_lengths[i];
-            let elem_size = elem_len + compact_size::encoded_size_const(elem_len as u64);
-            total_size += elem_size;
+            let encoded_size = compact_size::encoded_size_const(elem_len as u64) as u32;
+            let elem_size = elem_len.saturating_add(encoded_size);
+            total_size = total_size.saturating_add(elem_size);
             i += 1;
         }
         let witness_size = if !witness_element_lengths.is_empty() {
-            total_size + compact_size::encoded_size_const(witness_element_lengths.len() as u64)
+            let encoded_size =
+                compact_size::encoded_size_const(witness_element_lengths.len() as u64) as u32;
+            total_size.saturating_add(encoded_size)
         } else {
             0
         };
-        let script_size =
-            input_script_len + compact_size::encoded_size_const(input_script_len as u64);
+
+        let encoded_size = compact_size::encoded_size_const(input_script_len as u64) as u32;
```

Comment:
And here.

Reply:
As above, this is at most 9 so I fail to see how this loses data..